### PR TITLE
[TIMOB-23655] Implement Ti.Geolocation requestLocationPermissions() and hasLocationPermissions()

### PR DESCRIPTION
**Pull request by garymathews**
- Implement ``requestLocationPermissions()`` and ``hasLocationPermissions`` for ``Titanium.Geolocation``

###### TEST CASE
```Javascript
var win = Ti.UI.createWindow({backgroundColor: 'purple'}),
    btn_a = Ti.UI.createButton({top: '33%', title:'hasLocationPermissions()'}),
    btn_b = Ti.UI.createButton({top: '66%', title:'requestLocationPermissions()'});

btn_a.addEventListener('click', function () {
    alert('hasLocationPermissions() : ' + Ti.Geolocation.hasLocationPermissions(null));
});
btn_b.addEventListener('click', function () {
    Ti.Geolocation.requestLocationPermissions(null, function (res) {
        alert(JSON.stringify(res, null, ' '));
    });
});

win.add(btn_a);
win.add(btn_b);
win.open();
```

##### NOTE: ``Location`` capability must be present in ``tiapp.xml``. Also, ``requestLocationPermissions()`` is unsupported on Windows 8.1

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-23655)

### CODE REVIEW

**Comment by infosia** [comment]
> NOTE: Location capability must be present in tiapp.xml. Also, requestLocationPermissions() is unsupported on Windows 8.1

@garymathews  Do we have this note in the Geolocation doc? If not, please push it to [/apidoc/Titanium/Geolocation/Geolocation.yml](https://github.com/appcelerator/titanium_mobile/edit/master/apidoc/Titanium/Geolocation/Geolocation.yml) (http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Geolocation).

File: Source/Sensors/src/Geolocation.cpp [hunk 1]
```diff
@@ -64,6 +64,44 @@ namespace TitaniumWindows
...
+	bool Geolocation::hasLocationPermissions(const Titanium::Geolocation::AUTHORIZATION& authorizationType) TITANIUM_NOEXCEPT
+	{
+		// authorizationType is ignored, like on Android and <iOS8
+		return locationServicesAuthorization__ == Titanium::Geolocation::AUTHORIZATION::AUTHORIZED;
+	}
+
+	void Geolocation::requestLocationPermissions(const Titanium::Geolocation::AUTHORIZATION& authorizationType, JSObject callback) TITANIUM_NOEXCEPT
+	{
+		// this is not supported on Windows Phone 8.1
+		// permissions are granted at the time of app installation
```
Comment (infosia) [inline Source/Sensors/src/Geolocation.cpp]: We can detect it from AppxManifest, right? We actually load it in constructor.

[Geolocation.cpp#L215](https://github.com/appcelerator/titanium_mobile_windows/blob/master/Source/Sensors/src/Geolocation.cpp#L215)

So can we do like this?
```diff
+ #if !defined(IS_WINDOWS_10) 
+ locationServicesAuthorization__ = get_locationServicesEnabled() ? Titanium::Geolocation::AUTHORIZATION::AUTHORIZED : Titanium::Geolocation::AUTHORIZATION::DENIED;
+ #endif
```

Comment (garymathews) [inline Source/Sensors/src/Geolocation.cpp]: But then we wouldn't receive the permission request when calling ``requestLocationPermissions()``

Comment (infosia) [inline Source/Sensors/src/Geolocation.cpp]: This permission request only happens on Windows 10 (Mobile), right? I would think we want to set `Titanium::Geolocation::AUTHORIZATION::AUTHORIZED` on Windows Phone 8.1 too.

```diff
+#if defined(IS_WINDOWS_10) 
+	TitaniumWindows::Utility::RunOnUIThread([this, callback]() {
+	    concurrency::create_task(Geolocator::RequestAccessAsync()).then([this, callback](concurrency::task<GeolocationAccessStatus> task) {
+ ... skip ...
+#else
+   locationServicesAuthorization__ = get_locationServicesEnabled() ? Titanium::Geolocation::AUTHORIZATION::AUTHORIZED : Titanium::Geolocation::AUTHORIZATION::DENIED;
+#endif
```

Comment (garymathews) [inline Source/Sensors/src/Geolocation.cpp]: Yes! I'll amend the PR

**Comment by infosia** [comment]
test this please

**Pull request merged**